src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax;
+
+public class FigureSupplier extends ColorSupplier {
+    public static final int NUMBER_OF_FIGURES_NAME = 5;
+
+    public FigureSupplier() {
+
+    }
+
+    public String getRandomFigureName() {
+        return FigureName.values()[getRandom().nextInt(NUMBER_OF_FIGURES_NAME)].name();
+    }
+
+    public Figure getRandomFigure() {
+        return new Figure(getRandomColor(), getRandomFigureName());

```

Comment:
The `getRandomFigure` method seems to be creating a generic `Figure` object, which probably doesn't have a specific shape. You should create specific figure objects (e.g., `Circle`, `Square`, etc.) based on the random figure name.

Reply:
Fix comment